api-server/src/events/events.service.ts

```diff
@@ -56,6 +67,71 @@ export class EventsService {
     return this.eventModel.findAll(findOptions);
   }
 
+  async findAllPaginated({
+    tags = [],
+    verifiedOnly = true,
+    pageSize,
+    requestedPage,
+  }: {
+    tags: string[] | string;
+    verifiedOnly?: boolean;
+    pageSize: number;
+    requestedPage: number;
+  }): Promise<{ count: number; rows: EventModel[] }> {
+    return this.sequelize.transaction(async (_) => {
+      const tagClauseParams = this.getTagsClauseParams(tags);
+      const whereClause = this.getWhereClause(tagClauseParams, verifiedOnly);
+
+      // Sort events by the first start_time and apply pagination
+      const paginatedRows: EventModel[] = await this.sequelize.query(
+        `
+              with compressed_event as (SELECT events.*, min(dv.start_time) as first_start_time
+                                        FROM events
+                                                 LEFT OUTER JOIN datetime_venue dv on events.id = dv.event_id
+                  ${whereClause}
+              GROUP BY (events.id)
+              ORDER BY first_start_time DESC
+              OFFSET ${(requestedPage - 1) * pageSize} LIMIT ${pageSize} )
+              SELECT events.*
+              FROM compressed_event
+                       JOIN events ON events.id = compressed_event.id
+          `,
+        {
+          type: QueryTypes.SELECT,
+          model: EventModel,
+          replacements: {
+            tagFilter: tagClauseParams,
+          },
+        },
+      );
+
+      // Fill back in nested models date_times and venues
+      const dateTimes = await this.dateTimeVenueModel.findAll({
+        where: {
+          event_id: {
+            [Op.or]: paginatedRows.map(({ id }) => id),
+          },
+        },
```

Comment:
I don't know how Sequelize works under the hood, so maybe this isn't even possible, but would this make more sense as an `IN` condition, e.g. `event_id IN (id1, id2, ...)`?